backend/src/Edna/Upload/Service.hs

```diff
@@ -30,12 +43,22 @@ parseFile content =
 parseFile' :: FileContents -> Edna FileSummary
 parseFile' = measurementsToSummary . fcMeasurements
 
--- Need DB access for that
 compoundNameToId :: Text -> Edna (Maybe (SqlId Compound))
-compoundNameToId _ = pure Nothing
+compoundNameToId compoundName =
+  fmap (fmap mkSqlId) . runSelectReturningOne' $ select $ do
```

Comment:
Why don't you like `<<$>>`?

Reply:
Oh, right. Well, I don't like it for a couple of reasons: it's non-standard (people who are not used to `Universum` don't know this operator) and I always forget whether it's `<<$>>` or `<$$>`. And here there is an additional reason: it can't be used as an operator in infix position, it can be used only as a function and I slightly dislike that as well. So I think `fmap (fmap …` is ok here.